extensions/redis-client/deployment/src/main/java/io/quarkus/redis/client/deployment/DevServicesProcessor.java

```diff
@@ -0,0 +1,210 @@
+        if (!devServicesConfig.enabled) {
+            // explicitly disabled
+            log.debug("Not starting devservices for " + (isDefault(connectionName) ? "default redis client" : connectionName)
+                    + " as it has been disabled in the config");
+            return null;
+        }
+
+        String configPrefix = getConfigPrefix(connectionName);
+
+        boolean needToStart = !ConfigUtils.isPropertyPresent(configPrefix + RedisConfig.HOSTS_CONFIG_NAME);
+        if (!needToStart) {
+            log.debug("Not starting devservices for " + (isDefault(connectionName) ? "default redis client" : connectionName)
+                    + " as hosts have been provided");
+            return null;
+        }
+
+        DockerImageName dockerImageName = DockerImageName.parse(devServicesConfig.imageName.orElse(REDIS_6_ALPINE))
+                .asCompatibleSubstituteFor(REDIS_6_ALPINE);
+        FixedPortRedisContainer redisContainer = new FixedPortRedisContainer(dockerImageName, devServicesConfig.port);
+        redisContainer.start();
+        String redisHost = REDIS_SCHEME + redisContainer.getHost() + ":" + redisContainer.getPort();
+        return new StartResult(redisHost,
+                new Closeable() {
+                    @Override
+                    public void close() {
+                        redisContainer.close();
+                    }
+                });
+    }
+
+    private String getConfigPrefix(String connectionName) {
+        String configPrefix = QUARKUS + RedisConfig.REDIS_CONFIG_ROOT_NAME + DOT;
+        if (!isDefault(connectionName)) {
+            configPrefix = configPrefix + connectionName + DOT;
+        }
+        return configPrefix;
+    }
+
+    private static class StartResult {
+        private final String url;
+        private final Closeable closeable;
+
+        public StartResult(String url, Closeable closeable) {
+            this.url = url;
+            this.closeable = closeable;
+        }
+    }
+
+    private static class FixedPortRedisContainer extends GenericContainer {
+        OptionalInt fixedExposedPort;
+
+        public FixedPortRedisContainer(DockerImageName dockerImageName, OptionalInt fixedExposedPort) {
+            super(dockerImageName);
+            this.fixedExposedPort = fixedExposedPort;
+        }
+
+        @Override
+        protected void configure() {
+            super.configure();
+            if (fixedExposedPort.isPresent()) {
+                addFixedExposedPort(fixedExposedPort.getAsInt(), REDIS_EXPOSED_PORT);
+            } else {
+                addExposedPort(REDIS_EXPOSED_PORT);
+            }
+        }
+
+        public int getPort() {
+            if (fixedExposedPort.isPresent()) {
+                return fixedExposedPort.getAsInt();
+            }
+            return super.getFirstMappedPort();
+        }
+    }
+
+    static class DockerRunning extends IsDockerWorking {
```

Comment:
I think we should put a `IsDockerRunningSilent` class in `IsDockerRunning` (or besides it).

Reply:
Okay, let me put a separate commit for that.